### DISCUSSION TITLE
Its getting too many files with shadow endpoints living inside individual files

### DISCUSSION THREAD

**Discussion by yousufiqbal** [sveltejs/kit#3745] [General]
If there could be a way if shadow endpoints can be merged with .svelte file. It is more idiomatic I think. Since we already have individual files for standalone points and standalone endpoints cannot be replaced with shadow endpoints, since they have their own purpose.

Files living for both shadow endpoints and standalone endpoints makes projects with many files messy. I know there are downsides to merging it with .svelte file but these individual files are driving me crazy.

**Comment by benmccann** [comment]
You can try https://svemix.com/

**Reply by pixelmund** [reply]
Svemix uses shadow-endpoints now, i'm currently thinking about a way to get around creating them but i'm lost at that point.

**Reply by pixelmund** [reply]
Hey @benmccann i've created an issue inside the svemix repo for this:
[https://github.com/svemix/svemix/issues/50](https://github.com/svemix/svemix/issues/50)

I would really like to find a way to avoid generating the "Shadow"-Endpoints. With svemix the folders quickly gets really messy, but i can't think of a good way to do so. Do you or one of the other maintainers have any idea if or how we could solve this in Kit? Something like multiple routes-paths would probably help but idk.

**Comment by lkj4** [comment]
> shadow endpoints

What's a shadow endpoint?

**Reply by mustofa-id** [reply]
if page and endpoint have same file name in same folder, you can omit page load function, it called [shadow endpoints](https://github.com/sveltejs/kit/issues/3532).

**Reply by lkj4** [reply]
Oh thanks, wasn't aware of them and that they're already implemented in SvelteKit. So, we would we need Svemix anymore then?

**Reply by vekunz** [reply]
The YouTube channel LevelUpTuts has a good video that explains shadow endpoints https://youtu.be/PoYPZT7ruqI

**Comment by Antonio-Bennett** [comment]
You can still use the explicit load syntax if you want to

**Comment by JSaretin** [comment]
I don't think using shadow endpoint makes your projects messy, I am using sveltekit for most of my projects currently, and I haven't given it much of a thought, initially, I was using the load function to get data when the shadow endpoint wasn't working properly. if you think using shadow endpoint get messy, you might use a load function to get your data from one endpoint using a variable to differentiate your request like [this example](https://github.com/JSaretin/svelte-one-enpoint-to-mutiple-page) then again, you won't get the benefit of populating your template directly, (you would need to use props) overall, I think shadow endpoint is great, it makes me focus on a different part of my projects, like when dealing with user authentication or order creation. the ability to just export a variable on my template and populate it from the endpoint is the cherry on top.
